contracts/ColonyFunding.sol

```diff
@@ -106,7 +105,7 @@ contract ColonyFunding is ColonyStorage, PatriciaTreeProofs {
 
     if (_token == address(0x0)) {
       // Payout ether
-      msg.sender.transfer(remainder);
+      task.roles[_role].user.transfer(remainder);
```

Comment:
Just to be safe against sending funds into the abyss of 0x0, can we `assert` `task.roles[_role].user != address(0x0)`.

Reply:
Mm I'm not 100% on this for three reasons:

1) I'm not sure this is a state we can actually get into via the current Task flow. Why? Because for any user to receive a payout, the have to take some positive action. Managers setup the task; evaluators evaluate, workers submit work. So even _if_ there was a null account, it could not receive a positive payout.

2. There are many accounts other than `address(0x0)` which should not receive tokens. Adding this check gives a partial sense of security, in that it doesn't account for any of the countless possible addresses which shouldn't hold funds.

3. Because this branch is hard to test with the current Task flow (see point 1), it brings our coverage down below the 94% threshold and makes the build fail. 😭

Reply:
I disagree with all 3, to answer more specifically
1) The probability of funds being sent to an empty (vs non-empty) address combined with the fact that is an address we're guaranteed not to be able to recover funds from (as opposed to accidentally sending to another user for instance) increase the risk we fall into this scenario. Combined with the fact the ease of of this check makes it a recommended option. This is not misleading security but actually improved security of our contracts.
2) You've just described the very definition of when `assert` should be used, see the Solidity docs.
3) We have other invariants which are untestable, hence branch coverage is at 94% and not 100%. We should prefer to lower the `istanbul` coverage check instead of forsaking asserting the invariant.

Reply:
Otay